### PR TITLE
Support 0 value of Intel-specific RegisterAllocMode MD

### PR DESCRIPTION
**Pull request by sarnex**
We are planning on using this in the backend.

### CODE REVIEW

**Comment by sarnex** [comment]
@MrSidims Please review when you get a chance, I can't set the reviewers on here. Should be pretty non-interesting (hopefully)

File: test/transcoding/registerallocmode.ll [hunk 1]
```diff
@@ -6,16 +6,18 @@
 
 ; CHECK-SPIRV: 8 Decorate 4 UserSemantic "num-thread-per-eu 4"
 ; CHECK-SPIRV: 8 Decorate 6 UserSemantic "num-thread-per-eu 8"
-; CHECK-SPIRV-NOT: 8 Decorate 8
+; CHECK-SPIRV  8 Decorate 8 UserSemantic "num-thread-per-eu 0"
```
Comment (MrSidims) [inline test/transcoding/registerallocmode.ll:9]:
```suggestion
; CHECK-SPIRV:  8 Decorate 8 UserSemantic "num-thread-per-eu 0"
```

Comment (MrSidims) [reply test/transcoding/registerallocmode.ll:9]: Also, it seems like I've missed this during review of the initial patch.
Please do not add word count check of the instruction aka "8 Decorate 8" -> "Decorate 8". Also in remaining "Decorate 8" '8' is an ID of something, please explicitly add a check for this something and reuse its ID as a CHECK variable.

Comment (sarnex) [reply test/transcoding/registerallocmode.ll]: whoops, all fixed, thanks

**Review by MrSidims — APPROVED** [review]

**Pull request merged**